countryselector/src/commonTest/kotlin/Test.kt

```diff
@@ -0,0 +1,371 @@
+                                color = MaterialTheme.colorScheme.outline,
+                                shape = MaterialTheme.shapes.medium
+                            )
+                            .padding(8.dp),
+                        contentAlignment = Alignment.Center
+                    ) {
+                        CountrySelector(
+                            country = selectedCountry,
+                            onSelection = { selectedCountry = it },
+                            pickerRowContent = { country ->
+                                Row(
+                                    modifier = Modifier,
+                                    verticalAlignment = Alignment.CenterVertically
+                                ) {
+                                    Image(
+                                        painter = painterResource(
+                                            resource = country.flagImageResource,
+                                        ),
+                                        contentDescription = country.countryName,
+                                        modifier = Modifier
+                                            .size(24.dp)
+                                            .clip(CircleShape)
+                                    )
+                                    Spacer(Modifier.width(8.dp))
+                                    Text(country.countryName)
+                                    Spacer(Modifier.weight(1f))
+                                    Text(country.internationalDialCode)
+                                } },
+                            searchBarContent = { searchQuery, onQueryChange, hasError ->
+                                TextField(
+                                    value = searchQuery,
+                                    onValueChange = onQueryChange,
+                                    modifier = Modifier
+                                        .fillMaxWidth()
+                                        .padding(10.dp),
+                                    label = { Text("Search Countries") },
+                                    trailingIcon = {
+                                        if (hasError) {
+                                            Icon(Icons.Default.Warning, contentDescription = "No results found")
+                                        } },
+                                    isError = hasError,
+                                    placeholder = { Text("Search by country name or code") }
+                                )
+                            }
+                        )
+                    }
+                }
+
+                }
+
+            onNodeWithTag("TestBox").performClick() //ensure that the lang is changed
+            onNodeWithText("Search Countries").performTextInput("+20")
+            onNode(hasText("Égypte")).assertExists()
+            }
+        }
+
+    @Test
+    fun performenceTest(){
```

Comment:
Can you explain this test and give the method a more meaningful name so it's clear what's being tested?